ui/src/reducers/header.js

```diff
@@ -1,9 +1,33 @@
+import {
+  USER_REQUEST,
+  USER_SUCCESS,
+  USER_FAILURE
+} from '../actions/types.js';
+
 const initialState = {
-  currentUser: 'currentUser'
+  user: {
+    username: '',
+    email: ''
+  },
+  isUserRequestFailed: false
 };
 
 export default (state = initialState, action) => {
   switch (action.type) {
+    case USER_REQUEST:
```

Comment:
I suggest wrapping each case in a block, like this:

https://github.com/aerogear/mobile-security-service/blob/c0ca7cba530092ced9be67a9ed18fea150561ef2/ui/src/reducers/modals.js#L39

This allows variables to only exist in that case, allowing for reuse of variable names. We are in the process of adding this to the other reducers at the minute.